tools/SetupFlow/DevHome.SetupFlow/ViewModels/SummaryViewModel.cs

```diff
@@ -205,6 +217,31 @@ protected async override Task OnFirstNavigateToAsync()
             }
         }
 
+        // Collect all next steps.
+        var taskGroups = Orchestrator.TaskGroups;
+        foreach (var taskGroup in taskGroups)
+        {
+            var setupTasks = taskGroup.SetupTasks;
+            foreach (var setupTask in setupTasks)
+            {
+                if (setupTask.SummaryScreenInformation is not null &&
+                    setupTask.SummaryScreenInformation.HasContent)
+                {
+                    switch (setupTask)
+                    {
+                        case CloneRepoTask:
+                            var configResult = new CloneRepoSummaryInformationView();
+                            configResult.DataContext = setupTask.SummaryScreenInformation;
+                            _cloneRepoNextSteps.Add(configResult);
+                            break;
+                    }

```

Comment:
C# and MVVM is so complicated to me.  This code looks like Greek.

You're newing up a bunch of views that are 'next steps' in the clonerepo flow.

Shouldn't the next steps be simple small data structures that contain the minimal logical "next action data" - not views.  Then later when you need to show them, you create your views or user controls from those simple small data structures?

Reply:
Yeah.

After this comment I tried to use the view to use SummaryView.xaml to select that UI to show.  It is possible.  4 locations in DevHome do this.  Couldn't figure it out though.